hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -274,20 +315,25 @@ struct Vgen {
   void emit(const push& i);
   void emit(const roundsd& i);
   void emit(const sar& i);
+  void emit(const sarqi& i);
```

Comment:
This and the other shift vasm below are now implemented with an emitter instead of lowering them to the variations that do and don't set status flags. Those have been removed. They were initially created in an attempt to avoid expensive synthetic status flags setting. It's no longer needed since we're tracking precisely which status flags need to be set when emitting an instruction that defs SF.